lib/src/merged_tree.rs

```diff
@@ -1059,6 +1090,16 @@ impl Stream for TreeDiffStreamImpl<'_> {
     }
 }
 
+impl Stream for TreeDiffStreamImpl<'_> {
+    type Item = TreeDiffEntry;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        self.as_mut()
+            .poll_next_impl(cx)
+            .map(|o| o.map(|e| e.adjust_for_copy_tracking(&self.source_tree, self.copy_records)))
```

Comment:
nit: could you use some clearer names than `o` and `e` here?

Reply:
done